tools/meta_stdname_check.py

```diff
@@ -0,0 +1,357 @@
+#!/usr/bin/env python3
+
+"""
+
+This tool checks if all of the
+standard names present in a
+CCPP metadata file also exist
+in the standard names dictionary.
+
+The tool currently has two options:
+
+1.  A path to a single metadata file
+    is passed, in which case only that
+    file's standard names are checked, e.g.:
+
+./meta_stdname_check --metafile-loc /path/to/file.meta --stdname-dict /path/to/dict.xml
+
+2.  A path to a directory is passed, in
+    which case the directory is searched
+    for metadata files, and all found
+    files' standard names are checked, e.g.:
+
+./meta_stdname_check --metafile-loc /meta/path/ --stdname-dict /path/to/dict.xml
+
+"""
+
+######################################
+#Import needed standard python modules
+######################################
+
+import argparse
+import sys
+import os
+import os.path
+
+################################################
+#Add CCPP framework (lib) modules to python path
+################################################
+
+_CURR_DIR = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(os.path.join(_CURR_DIR, "lib"))
+
+#######################################
+#Import needed framework python modules
+#######################################
+
+from xml_tools import read_xml_file
+
+#################
+#Helper functions
+#################
+
+#++++++++++++++++++++++++++++++
+#Input Argument parser function
+#++++++++++++++++++++++++++++++
+
+def parse_arguments():
+
+    """
+    Parses command-line input arguments
+    using the argparse python module and
+    outputs the final argument object.
+    """
+
+    #Create description:
+    desc = "Check if the metafile contains variable standard names\n"
+    desc += "that are not in the provided standard names dictionary."
+
+    #Create parser object:
+    parser = argparse.ArgumentParser(description=desc)
+
+    #Add input arguments to be parsed:
+    parser.add_argument('--metafile-loc',
+                        metavar='<path to directory or file>',
+                        action='store', type=str,
+                        help="Location of metadata file(s)")
+
+    parser.add_argument('--stdname-dict',
+                        metavar='<path to file>',
+                        action='store', type=str,
+                        help="Location of standard name dictionary (XML file)")
```

Comment:
```suggestion
    parser.add_argument('-m', '--metafile-loc',
                        metavar='<path to directory or file>',
                        action='store', type=str,
                        help="Location of metadata file(s)")

    parser.add_argument('-s', '--stdname-dict',
                        metavar='<path to file>',
                        action='store', type=str,
                        help="Location of standard name dictionary (XML file)")
```

Just a personal preference suggestion, I'm a big fan of short option names.

Reply:
Good idea!  The suggested short option names have now been added.